src/frontend/src/instance.rs

```diff
@@ -411,6 +408,14 @@ impl Instance {
             .context(error::ShutdownServerSnafu)
             .map(|_| ())
     }
+
+    #[cfg(test)]
+    pub(crate) fn is_distributed_instance(&self) -> bool {
```

Comment:
It's really tricky. I think we can just add a `bool` value to represent its mode.